### PR TITLE
Add Arm CPU capabilities environment variable override

### PR DESCRIPTION
**Pull request by dkostic**
### Issues:
N/A

### Description of changes: 
Add Arm CPU capabilities environment variable override.

### Call-outs:
The effect of the change can be checked for example by comparing the speed of SHA-256 when run with `bssl speed SHA-256` and `export OPENSSL_armcap_P=0 && bssl speed SHA-256`.

### Testing:
How is this change tested (unit tests, fuzz tests, etc.)? Are there any testing steps to be verified by the reviewer?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: crypto/cpu-aarch64-linux.c [hunk 1]
```diff
@@ -57,6 +95,13 @@ void OPENSSL_cpuid_setup(void) {
   if (hwcap & kSHA256) {
     OPENSSL_armcap_P |= ARMV8_SHA256;
   }
+
+  const char *env;
+  env = getenv("OPENSSL_armcap_P");
+  if (env == NULL) {
+    return;
+  }
+  handle_cpu_env(&OPENSSL_armcap_P, env);
```
Comment (DominicDams) [inline crypto/cpu-aarch64-linux.c:104]: Maybe instead we should do:

```
if (env != NULL) {
    handle_cpu_env(&OPENSSL_armcap_P, env);
}
```

For additional clarity

**Review by DominicDams — APPROVED** [review]

File: crypto/cpu-aarch64-linux.c [hunk 2]
```diff
@@ -19,13 +19,51 @@
...
+#include <string.h>
+
 #include <openssl/arm_arch.h>
 
 #include "internal.h"
 
 
 extern uint32_t OPENSSL_armcap_P;
 
+// handle_cpu_env applies the value from |in| to the CPUID values in |out[0]|
```
Comment (nebeid) [inline crypto/cpu-aarch64-linux.c:38]: Could this comment include example usage?
To set up 2 capabilities, how is it done? I'm guessing by putting directly the value that represents both.
How can `~` and `|` be used?

**Review by darylmartin100 — DISMISSED** [review]

File: crypto/cpu-aarch64-linux.c (hunk 2 above)
Comment (dkostic) [reply crypto/cpu-aarch64-linux.c:38]: done

File: crypto/cpu-aarch64-linux.c (hunk 1 above)
Comment (dkostic) [reply crypto/cpu-aarch64-linux.c]: done

**Review by darylmartin100 — APPROVED** [review]

**Review by DominicDams — APPROVED** [review]

**Pull request merged**